### ISSUE TITLE
 Number count of errors in `velero restore get` is incorrect

### ISSUE THREAD

**Issue by qpdpQ** [vmware-tanzu/velero#7563]
**What steps did you take and what happened:**

When I restore resources, data is not restored and I can see there is some error message in velero pod:
```
time="2024-03-15T09:43:57Z" level=error msg="Error executing hook" error="command terminated with exit code 1" hookPhase=post hookSource=annotation hookType=exec logSource="/remote-source/velero/app/internal/hook/wait_exec_hook_handler.go:173" pod=bawent/mongodb-backup-84bc8b8cb5-d9k2g restore=velero/restore-mongo-data
```
but the number count of ERRORS in `velero restore get` is 0
```
NAME                     BACKUP        STATUS      STARTED                         COMPLETED                       ERRORS   WARNINGS   CREATED                         SELECTOR
restore-mongo-data       backup-2302   Completed   2024-03-15 09:42:05 +0000 UTC   2024-03-15 09:43:57 +0000 UTC   0        0          2024-03-15 09:42:05 +0000 UTC   foundationservices.cloudpak.ibm.com=mongo-data
```

**What did you expect to happen:**
The `velero restore get` should show the number of errors correctly

**The following information will help us better understand what's going on**:

[bundle-2024-03-25-11-24-39.tar.gz](https://github.com/vmware-tanzu/velero/files/14747993/bundle-2024-03-25-11-24-39.tar.gz)

**Environment:**

- Velero version (use `velero version`):
```
# velero version
Client:
        Version: v1.12.3
        Git commit: 684f71306e9c2fda204a16cb012dc209523cfae1
Server:
        Version: v1.9.5-OADP
``` 
- Velero features (use `velero client config get features`): 
```
# velero client config get features
features: <NOT SET>
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.2", GitCommit:"846602e50eb29ecdc7441ebed2fc846048959eaa", GitTreeState:"clean", BuildDate:"2023-03-08T15:10:20Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.16+6df2177", GitCommit:"f5b7c3e8faedd51935d77828a5fc72c7540236f4", GitTreeState:"clean", BuildDate:"2024-01-18T12:53:29Z", GoVersion:"go1.19.13 X:strictfipsruntime", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
- Cloud provider or hardware configuration: 
```
Openshift Cloud Platform
Client Version: 4.12.8
Kustomize Version: v4.5.7
Server Version: 4.12.50
Kubernetes Version: v1.25.16+6df2177
```
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by allenxu404** [comment]
Thanks for pointing it out. 

This issue will be fixed with this PR: https://github.com/vmware-tanzu/velero/pull/7619 merged.

**Comment by allenxu404** [comment]
The fix will be included in v1.14

**Comment by reasonerjt** [comment]
Closing as fixed.
